### ISSUE TITLE
Error : RHS does not allow direction 'ORTHOGONAL' (on line that doesn't contain the keyword)

### ISSUE THREAD

**Issue by increpare** [bvoq/mixedaim#1]
https://www.puzzlescript.net/editor.html?hack=4d31732deeb111799c1d2da630ac7f17

works ok in puzzlescript but PSMIS gives the error `RHS does not allow direction 'ORTHOGONAL'` for the line

```
[moving wall bond ] -> [ moving wall moving bond ]
```

**Comment by increpare** [comment]
and if I remove this line (or replace it with the equivalent

```
[up wall bond ] -> [ up wall up bond ]
+ [down wall bond ] -> [ down wall down bond ]
+ [left wall bond ] -> [ left wall left bond ]
+ [right wall bond ] -> [ right wall right bond ]
```

it gets freaked out in the same way by  `[vertical player_h ] -> [ vertical player_v ]`

I guess this is a difference in rule inference?

**Comment by bvoq** [comment]
Yes, at the moment I don't allow vertical/moving on the RHS. Will need to look into that.

**Comment by increpare** [comment]
(FWIW/IIRC the direction deduction algorithm is exactly the same as the
property deduction one)

Am Sa., 13. Juli 2019 um 23:04 Uhr schrieb Kevin De Keyser <
notifications@github.com>:

**Comment by bvoq** [comment]
I've updated the error message, but haven't come around to fixing it since it yet.
It can be easily changed to a group by the user himself.
